driver-core/src/test/java/com/datastax/driver/core/RollingCountTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.mockito.Mockito.when;
+
+public class RollingCountTest {
+    @Mock
+    com.codahale.metrics.Clock clock;
```

Comment:
The metrics Clock interface is pretty simple, and we are only using 1 method (`getTick`).  You brought this up before, but what do you think about replacing this with our own Clock?

Thinking that either we could extend `com.datastax.driver.core.Clock` to have 2 methods (for realtime and monotonic time), or a separate package private Clock for monotonic time.